src/controllers/details.ts

```diff
@@ -25,15 +27,15 @@ export async function getAppVersions(req, res) {
   }
 }
 
-export const checkRoute = async (req, res) => {
+export const checkRoute = async (req: Request, res: Response) => {
   if (!req.owner) return failure(res, 'no owner')
 
   const { pubkey, amount, route_hint } = req.query
   if (!(pubkey && pubkey.length === 66)) return failure(res, 'wrong pubkey')
 
   const owner = req.owner
   try {
-    const amt = parseInt(amount) || constants.min_sat_amount
+    const amt = parseInt((amount || '').toString()) || constants.min_sat_amount
```

Comment:
Same as above one but maybe we can create a new type that is based off `Request` but includes `amount` as a number